src/rHelpProviderBuiltin.ts

```diff
@@ -0,0 +1,103 @@
+
+
+
+import * as cp from 'child_process';
+
+import * as http from 'http';
+
+import * as rHelpPanel from './rHelpPanel';
+
+export interface RHelpClientOptions extends rHelpPanel.RHelpProviderOptions {
+	// path of the R executable. Could be left out (with limited functionality)
+    rPath: string;
+}
+
+
+// Class to forward help requests to a backgorund R instance that is running a help server
+export class RHelpClient implements rHelpPanel.HelpProvider {
+    private cp: cp.ChildProcess;
+    private port: number|Promise<number>;
+    private readonly rPath: string;
+
+    public constructor(options: RHelpClientOptions){
+        this.rPath = options.rPath || 'R';
+        this.port = this.launchRHelpServer(); // is a promise for now!
+    }
+
+    public async launchRHelpServer(){
+        // starts the background help server and waits forever to keep the R process running
+        const cmd = (
+            `${this.rPath} --silent --no-echo --vanilla -e ` +
+            `"cat(tools::startDynamicHelp(),'\\n'); while(TRUE) Sys.sleep(1)" ` 
```

Comment:
"--no-echo" is a replacement of "--slave" in R >4.0.0. Note that "--slave" can still be used.

I see three options:

1. Simply use "--slave" instead of "--no-echo" -> this will cause problems when "--slave" gets deprecated.  
2. Extract somehow the version of R: E.g., run `${this.rPath} --version` first, extract the R version from that string, and then start the R help session with the appropriate switch.   
3. Drop the "--no-echo"/"--slave" switch, and extract the port number using a regular expression from the stdout string.

Reply:
Thanks for finding this, I went with option 1 for now (we can always implement more sophisticated methods, if/when they deprecate `--slave` at some point)